tests/testHelpers/buildStateTransitionHelpers.ts

```diff
@@ -0,0 +1,202 @@
+import { expect } from 'chai';
+import { ScenarioRegistrant } from '../../semantic-mocha/src';
+import { RoomHandler } from '../../src/roomHandlers/roomHandler';
+import { DeveloperError } from '../../src/utils/developerError';
+import {
+  CommandResult,
+  DEFAULT_COMMAND,
+  NarrowedRoomState,
+  RoomType,
+  isNil,
+} from '../../src/utils/types';
+import {
+  generateNarrowedRoomState,
+  NarrowedRoomStateOverride,
+} from './generateGameState';
+import { INVALID_COMMAND } from './invalidCommand';
+
+type StateTransitionHelpers<TRoomType extends RoomType> = {
+  testDefaultCommandAtEntrance: () => void;
+  testInvalidCommandAtEntrance: () => void;
+  testStateTransition: StateTransitionScenarioRegistrant<TRoomType>;
+};
+
+type StateTransitionScenarioRegistrant<TRoomType extends RoomType> = (
+  mermaidTransition: string,
+  onArrange: OnArrangeAccessor<TRoomType>,
+) => void;
+
+type OnArrangeAccessor<TRoomType extends RoomType> =
+  () => ArrangedTransitionData<TRoomType>;
+
+export type ArrangedTransitionData<TRoomType extends RoomType> = {
+  startingRoomState: OmitKnownKeys<TRoomType>;
+  expectedResult: {
+    commandDescription: CommandResult<TRoomType>['commandDescription'];
+    roomState: null | OmitKnownKeys<TRoomType>;
+  };
+};
+
+const EXIT_CODE = '[*]';
+
+// Have to exclude room states that only have "type" and "playerState" or they become "any"
+type OmitKnownKeys<TRoomType extends RoomType> = TRoomType extends
+  | 'exampleRoom1'
+  | 'exampleRoom2'
+  ? Record<string, never>
+  : Omit<NarrowedRoomState<TRoomType>, 'type' | 'playerState'>;
+
+export const buildStateTransitionHelpers = <TRoomType extends RoomType>(
+  testScenario: ScenarioRegistrant,
+  roomType: TRoomType,
+  roomHandler: RoomHandler<TRoomType>,
+): StateTransitionHelpers<TRoomType> => {
+  const testStateTransition: StateTransitionScenarioRegistrant<TRoomType> = (
+    mermaidTransition: string,
+    onArrange: OnArrangeAccessor<TRoomType>,
+  ) => {
+    testScenario(mermaidTransition)
+      .arrange(() => {
+        const match = mermaidTransition.match(
+          /^([^\s]*) --> ([^\s]*): ([^\s]*)$/,
+        );
+
+        const [, startingPlayerState, endingPlayerState, command] = match ?? [
+          null,
+          null,
+          null,
+          null,
+        ];
+
+        // Validates values instead of "match" to eliminate "undefined" as a possibility
+        if (
+          isNil(startingPlayerState) ||
+          isNil(endingPlayerState) ||
+          isNil(command)
+        ) {
+          throw new DeveloperError(
+            'Malformed mermaid transition code. Check for extra or missing whitespace, or a deviance from the expected REGEX',
```

Comment:
This could new custom markdownlint rule. (don't need to add it to the existing PR, maybe a good idea for the backlog)

Reply:
I had a similar idea to lint the mermaid code that is passed to `testStateTransition` I made a ticket for it.